packages/dds/merge-tree/src/properties.ts

```diff
@@ -69,14 +69,24 @@ export function combine(
 }
 
 export function matchProperties(a: PropertySet | undefined, b: PropertySet | undefined) {
+	let entriesA;
+	let entriesB;
+	if (!a && !b) {
+		return true;
+	}
```

Comment:
You could probably simplify this function by defaulting to an empty list and then doing a length check. This lets you avoid the second `for` loop over `b`'s keys and removes a lot of the nested `if` stmts.

I wouldn't be able to comment on whether this is faster, but it does seem like the more intuitive implementation to me.

```ts
if (!a && !b) {
  return true;
}

const keysA = a ? Object.keys(a) : [];
const keysB = b ? Object.keys(b) : [];

if (keysA.length !== keysB.length) {
  return false;
}

for (const key of keysA) {
	if (b?.[key] === undefined) {
		return false;
	} else if (typeof b[key] === "object") {
		if (!matchProperties(a?.[key], b[key])) {
			return false;
		}
	} else if (b[key] !== a?.[key]) {
		return false;
	}
}

return true;
```